src/blog/2023/07/how-to-deploy-a-basic-opc-ua-server-in-node-red.md

```diff
@@ -0,0 +1,686 @@
+flow.set('isoInput7', Math.random() + 17.0)
+flow.set('isoInput8', Math.random() + 18.0)
+
+msg.payload = [
+    flow.get('isoInput1'),
+    flow.get('isoInput2'),
+    flow.get('isoInput3'),
+    flow.get('isoInput4'),
+    flow.get('isoInput5'),
+    flow.get('isoInput6'),
+    flow.get('isoInput7'),
+    flow.get('isoInput8'),
+]
+return msg;
+```
+- another inject node is triggering the function `set flow context Outputs`, also at a one second interval, which creates another set of 7 randomly generated float values and stores them as flow context variables, `isoOutput2` - `isoOutput8` (isolated inputs).  The values will change to a new random number each time the node is injected.
+
+```javascript 
+// flow.set('isoOutput1', Math.random() + 1.0) interval comes from server
+flow.set('isoOutput2', Math.random() + 2.0)
+flow.set('isoOutput3', Math.random() + 3.0)
+flow.set('isoOutput4', Math.random() + 4.0)
+flow.set('isoOutput5', Math.random() + 5.0)
+flow.set('isoOutput6', Math.random() + 6.0)
+flow.set('isoOutput7', Math.random() + 7.0)
+flow.set('isoOutput8', Math.random() + 8.0)
+
+msg.payload = [
+    flow.get('isoOutput1'),
+    flow.get('isoOutput2'),
+    flow.get('isoOutput3'),
+    flow.get('isoOutput4'),
+    flow.get('isoOutput5'),
+    flow.get('isoOutput6'),
+    flow.get('isoOutput7'),
+    flow.get('isoOutput8'),
+]
+return msg;
+```
+
+We can confirm the values are being stored in memory by checking the flow context data and pressing the refresh button.
+
+![Context Data 1](./images/opc-ua-1/context-data-1.png)
+![Context Data 2](./images/opc-ua-1/context-data-2.png)
+Each time we hit refresh, the values change, confirming that the values are randomly changing every second.
+
+The last, and most important part of the flow, is the `Compact-Server` node, which actually stands alone without any incoming or outgoing connections.
+
+![Compact Server Node](./images/opc-ua-1/compact-server-node.png)
+In the `Compact-Server` node properties, the first tab is `Settings`, and the two important properties here are `Port` and `Show Errors`.  As can be seen in the node screenshot above, the node is reporting `active`, which means the server is configured correctly.  
+
+![Settings Tab](./images/opc-ua-1/settings-tab.png)
+
+The `Limits` tab specifies some default limits that we can configure if we like, but are not necessary to be modified for test purposes.  
+
+The `Security` tab has one important option, `Allow Anonymous`.  By default, anonymous access is enabled.  
+
+![Security Tab](./images/opc-ua-1/security-tab.png)
+For a production system, we will want to enable security, but for test purposes, we will leave anonymous access enabled.  
+
+`Users & Sets` tab is related to security and permissions.  We can leave this empty for testing.  
+
+ The `Address Space` tab is where the information for constructing the OPC Information Model is stored, and consists of a function that is responsible for invoking the OPC-UA server, 
+
```

Comment:
I find the following section with pages of javascript to be hard to read and I am not really sure the value of having it here? If someone wants to read it can't they just go to this flow and read it themselves?

Reply:
I agree, when I read it back it's way too much js.  But I don't want completely remove it. 
The js code in the address space is over 260 lines long, and the code isn't commented or explained in the example.  So explaining each section of code is my way of "commenting the code".  So I will shorten the code to snippets, rather than pasting the whole thing.  Because I do agree, they can get the full code from the example.